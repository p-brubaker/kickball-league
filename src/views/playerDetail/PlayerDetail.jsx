import { useParams } from 'react-router-dom'
import { useState, useEffect } from 'react'
import { getPlayerById } from '../../services/players'
import { getTeamById } from '../../services/teams'

export default function PlayerDetail() {
    const { id } = useParams()
    const [player, setPlayer] = useState(null)

    useEffect(() => {
        const get = async () => {
            const player = await getPlayerById(id)
            const team = await getTeamById(player.team_id)
            player.team = team.name
            setPlayer(player)
        }
        get()
    },[id])

    return (
        player ? (
        <div className="player-detail">
            <p>Name:{` ${player.name}`}</p>
            <p>Position:{` ${player.position}`}</p>
            <p>Team:{` ${player.team}`}</p>

        </div>
    ) : <p>Loading...</p>
    )
}
