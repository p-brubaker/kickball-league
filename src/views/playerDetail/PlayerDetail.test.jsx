import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router'
import PlayerDetail from './PlayerDetail'
import { setupServer } from 'msw/node'
import { rest } from 'msw'

const mockPlayer = {
    id: 10,
    created_at: '2021-12-12T05:34:30.551Z',
    name: 'Bud Hayes',
    team_id: 1,
    position: 'Designated Hitter'
}

const mockTeam = {
    id: 1,
    created_at: '2021-12-12T05:34:30.551Z',
    
}

const server = setupServer(
    rest.get('https://paozlutduarjgvzaiduh.supabase.co/rest/v1/players',
        (req, res, ctx) => {
            return res(ctx.json([mockPlayer]))
        }
    ),
    rest.get('https://paozlutduarjgvzaiduh.supabase.co/rest/v1/teams',
        (req, res, ctx) => {
            return res(ctx.json([mockTeam]))
        }
    )
)

beforeAll(() => {
    server.listen()
})

afterAll(() => {
    server.close()
})

it('should render player details', async () => {
    const { container } = render(
        <MemoryRouter>
            <PlayerDetail match={{params: { id: 1 }}} />
        </MemoryRouter>
    )

    screen.getByText(/Loading.../)

    await screen.findByText(/Name:/)
    screen.getByText(/Position:/)
    screen.getByText(/Team:/)
    expect(container).toMatchSnapshot()

})
