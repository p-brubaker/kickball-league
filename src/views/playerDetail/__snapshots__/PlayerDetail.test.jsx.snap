// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render player details 1`] = `
<div>
  <div
    class="player-detail"
  >
    <p>
      Name:
       undefined
    </p>
    <p>
      Position:
       undefined
    </p>
    <p>
      Team:
       undefined
    </p>
  </div>
</div>
`;
